src/nvim/terminal.c

```diff
@@ -496,37 +495,33 @@ static int terminal_execute(VimState *state, int key)
 void terminal_destroy(Terminal *term)
 {
   buf_T *buf = handle_get_buffer(term->buf_handle);
-  if (buf) {
-    term->buf_handle = 0;
-    buf->terminal = NULL;
-  }
+  assert(buf);
+  apply_autocmds(EVENT_TERMCLOSE, NULL, NULL, false, buf);
```

Comment:
Yes, the documentation is not accurate there. However, if we wait for the job to terminate or get killed, `TermClose` may be triggered after the buffer was wiped, meaning that autocommands are no longer able to do things with the buffer content. On the other hand, when the job dies but the buffer stays, we still want to generate the event. How would you suggest to go forward with this?